omegaconf/_utils.py

```diff
@@ -648,16 +649,19 @@ def _valid_dict_key_annotation_type(type_: Any) -> bool:
     return type_ is None or type_ is Any or issubclass(type_, DictKeyType.__args__)  # type: ignore
 
 
+BASE_TYPES = (
+    int,
+    float,
+    bool,
+    bytes,
+    str,
+    type(None),
+)
+
+
 def is_primitive_type(type_: Any) -> bool:
     type_ = get_type_of(type_)
-    return issubclass(type_, Enum) or type_ in (
-        int,
-        float,
-        bool,
-        str,
-        bytes,
-        type(None),
-    )
+    return issubclass(type_, (Enum, pathlib.Path)) or type_ in BASE_TYPES
```

Comment:
I think checking `issubclass(type_, pathlib.Path)` here is not the best approach.
We don't want to support custom user-defined subclasses of `Path` (i.e. `class MyPath(pathlib.Path): ...` should not be supported).

Reply:
the problem is that `pathlib.Path` isn't an actual class. The classes are `PosixPath` and `WindowsPath`

Reply:
Oh, I see. Thanks for the clarification.